### DISCUSSION TITLE
Iframe component json format

### DISCUSSION THREAD

**Discussion by shelendravashishtha2** [GrapesJS/grapesjs#4843] [Q&A]
Hi @artf ,

I'm using grapesjs json to render data on my editor using

`editor.setComponents();
editor.setStyle()`

functions. now when I'm trying to add these components inside div i can provide the custom json like 
`
{
         "tagName: "div",
          attributes: {
          },
          components : componentArray
}
`
but same json is not working in case of iframe. Iframe is getting rendered on the editor but  components are getting rendered outside the body of the document of iframe.

Iframe Json is like 
`
{
         "tagName: "iframe",
          attributes: {
          },
          components : componentArray
}
`
Can you please help me with this ?

**Comment by artf** [comment]
iframes don't work in that way, you generally pass the `src`, `srcdoc` or inject the content via JS (if it's not a cross-origin iframe).
What is your use case of using an iframe in that way?

**Reply by shelendravashishtha2** [reply]
actually I want to render multiple pages on single editor instance and for that I'm injecting all the pages in the form of json by using above functions. 

but the problem is that when I'm using other tags it the style of of the pages are getting applied to the editor itself and the screen is getting messed up because of that.

so just to resolve this issue  I'm trying to generate my page json in iframe document container so that all the styles will be applied to  document of the page itself and not just on complete editor.

**Reply by shelendravashishtha2** [reply]
Hi @artf 
any suggestions .... ?

**Reply by artf** [reply]
You can have multiple pages on a single editor instance https://codepen.io/artf/pen/XWpJQoY

**Reply by shelendravashishtha2** [reply]
Actually, I want to show all the pages on a single screen ,not by just showing the page name in side bar on a single editor instance. like multiple pages can be edited and seen in figma. So for that I wanted to use iframe. Even if I render the components manually, I am not able to drag and drop inside the iframe src?

**Reply by shelendravashishtha2** [reply]
![Untitled](https://user-images.githubusercontent.com/53626635/213185894-65f05916-c397-4931-97fb-d15f595f5adc.jpg)
for the reference this should be the output 
and those pages should be editable

**Reply by artf** [reply]
I've started to work on [something like that](https://github.com/artf/grapesjs/pull/2524) but ,unfortunately, due to lack of time I left this feature behind.
